ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.js

```diff
@@ -117,11 +118,18 @@ const ConfirmAddSuggestedToken = (props) => {
     );
   }, [suggestedAssets, tokens, t]);
 
+  useEffect(() => {
+    if (Object.keys(addedToken).length) {
```

Comment:
This sustains the behaviour of tracking the metrics event when the submit button is clicked, but it does so indirectly. Previously, the metrics event would be tracked as part of the onclick method call, but not it is tracked as a result of a state change that is created by the onclick method call.

This could cause confusion for a reader of the code, who might expect `useEffect` to be responding to a state change that is outside of the control of this component. Also, it increases the number of places in which a bug could be introduced in the future. Previously, the path was `onclick call -> metric event method call`, now the path is `onclick call -> state update -> useEffect recognition of state update -> metric event method call`. So there are more potential steps that could be accidentally broken in the future. Finally, as a general rule, if useEffect can be avoided for something simpler, that generally make components more readable and maintainable.

And in this case, there is some risk that this might not always work as desired when there are multiple `suggestedAssets` 

I would suggest that we continue just to call the metrics event method within the onclick handler. To do this, you could put the whole onclick handler function in a function defined in the body of the component function and then pass that to `onClick`.

I think this would require using the existing `tokenAddedEvent` method

Reply:
@danjm 

Thank you for calling this out! As mentioned in this past Wednesday's standup, I have updated the logic to call `trackEvent` instead of `useNewMetricEvent` directly. 

I agree using `useEffect` can cause unwanted side-effects as `suggestedAssets` is susceptible to change. 

I did go with calling the click handler from the `onClick` function. I think the biggest hurdle was finding a way to use `useNewMetricEvent`, and the key was not to use this when we have dynamic params.  Since `useNewMetricEvent` returns a custom hook, we cannot call it inside another callback which would cause some limitations.